Description: don't display originator table if primary interface is inactive
Origin: backport, http://open-mesh.net/changeset/1324
Author: Marek Lindner <lindner_marek@yahoo.de>

---
diff --git a/proc.c b/proc.c
index 79c21486ae5d725c622fa98392dd3f89b5ff5e93..21d6e36cf813322c6aed44315484d31b15d6b24e 100644
--- a/proc.c
+++ b/proc.c
@@ -389,6 +389,12 @@ int proc_originators_read(struct seq_file *seq, void *offset)
 		goto end;
 	}
 
+	if (((struct batman_if *)if_list.next)->if_active != IF_ACTIVE) {
+		rcu_read_unlock();
+		seq_printf(seq, "BATMAN disabled - primary interface not active \n");
+		goto end;
+	}
+
 	seq_printf(seq, "  %-14s (%s/%i) %17s [%10s]: %20s ... [B.A.T.M.A.N. Adv %s%s, MainIF/MAC: %s/%s] \n", "Originator", "#", TQ_MAX_VALUE, "Nexthop", "outgoingIF", "Potential nexthops", SOURCE_VERSION, (strlen(REVISION_VERSION) > 3 ? REVISION_VERSION : ""), ((struct batman_if *)if_list.next)->dev, ((struct batman_if *)if_list.next)->addr_str);
 
 	rcu_read_unlock();
